app.js

```diff
@@ -62,29 +62,281 @@ if (sessionStorage.getItem('SMART_KEY')) { // is there an event like FHIR.oauth2
+        const AllergyIntoleranceContent = document.getElementById('AllergyIntoleranceContent');
+        const MedicationStatementContent = document.getElementById('MedicationStatementContent');
+        const MedicationRequestContent = document.getElementById('MedicationRequestContent');
+        const MedicationContent = document.getElementById('MedicationContent');
+        const ConditionContent = document.getElementById('ConditionContent');
+        const ObservationContent = document.getElementById('ObservationContent');
+        const OrganizationContent = document.getElementById('OrganizationContent');
+        const DeviceContent = document.getElementById('DeviceContent');
+        const DeviceUseStatementContent = document.getElementById('DeviceUseStatementContent');
+        const DiagnosticReportContent = document.getElementById('DiagnosticReportContent');
+        const ImagingStudyContent = document.getElementById('ImagingStudyContent');
+        const MediaContent = document.getElementById('MediaContent');
+        const PractitionerContent = document.getElementById('PractitionerContent');
+        const PractitionerRoleContent = document.getElementById('PractitionerRoleContent');
+        const ProcedureContent = document.getElementById('ProcedureContent');
+        const SpecimenContent = document.getElementById('SpecimenContent');
 
-        client.request('Patient/' + client.getPatientId()).then(patient => {
+        const requestAndDisplay = (resourceType, displayCallback) => {
+            endpoint = (resourceType == 'Patient' ? 'Patient/' : `${resourceType}?patient=`) + client.getPatientId()
+            client.request(endpoint).then(resources => {
+                resourcesToPass = []
+                resources.forEach(resource => {
+                    if (resource === undefined || resource.resourceType != resourceType) return;
+                    resourcesToPass.push(resource);
+                });
+                displayCallback(resourcesToPass);
+            });
+        }
+
+        requestAndDisplay('Patient', patient => {
             const name = patient.name[0];
             const formattedName = `${name.given.join(' ')} ${name.family}`;
-            patientInfo.innerHTML = `<h2>Patient Name: ${formattedName}</h2>`;
+            patientInfo.append(`<h2>Patient Name: ${formattedName}</h2>`);
         });
 
-        client.request(`Immunization?patient=${client.getPatientId()}`, { flat: true }).then(immunizations => {
-            immunizationHistory.innerHTML = '<h2>Immunization History:</h2>';
+        requestAndDisplay('Immunization', immunizations => {
+            immunizationHistory.append('<h2>Immunization History:</h2>');
             const list = document.createElement('ul');
 
             for (let i = 0; i < immunizations.length; i++) {
                 const immunization = immunizations[i];
-                if (immunization === undefined || immunization.resourceType != 'Immunization') continue;
```

Comment:
I needed these checks, see https://github.com/uwcirg/sof_uscdi_app/commit/902c6aa1fc8be64badba576c3ff51eafa69fa4b0

Reply:
No problem, it's just [pulled out into `requestAndDisplay`](https://github.com/uwcirg/sof_uscdi_app/blob/26476da145d9dc72a51c52afe74f648d340bb8de/app.js#LL87C22-L87C22) so it applies to each of these requests.